core/src/com/unciv/logic/city/managers/CityPopulationManager.kt

```diff
@@ -160,11 +160,15 @@ class CityPopulationManager : IsPartOfGameInfoSerialization {
             .filter { it.getOwner() == currentCiv && !it.isBlockaded() }.toList().asSequence()
 
         val localUniqueCache = LocalUniqueCache()
+        val rankTileForCityWorkCache = mutableMapOf<Tile, Float>() // Create the tile cache
+
         for (i in 1..getFreePopulation()) {
             //evaluate tiles
             val (bestTile, valueBestTile) = tilesToEvaluate
                     .filterNot { it.providesYield() }
-                    .associateWith { Automation.rankTileForCityWork(it, city, cityStats, localUniqueCache) }
+                    .associateWith { rankTileForCityWorkCache.getOrPut(it) {
+                        Automation.rankTileForCityWork(it, city, cityStats, localUniqueCache)
+                    } }
```

Comment:
Uncacheable, since rank tile *for city work* depends on how much the city needs food - and the more population we assign to food tiles, the less they become important, so this caching actually disrupts

Reply:
Will revert... Thanks for catching. Just FTR this is super heavy, too and ends up in the updateStats thing I mentioned in the other report again:

![Screenshot from 2023-04-24 22-33-01](https://user-images.githubusercontent.com/126110113/234110229-8f1c5420-f055-475e-9f88-c30b2e9f7348.png)

Reply:
Can you drill down more into the function to see why? Chances are there's stuff we can cache in that function if it's heavy